main.go

```diff
@@ -149,6 +157,42 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Subscription controller - skipping subscription controller if preflights are bypassed.
+	if !helper.IsPreflightBypassed() {
```

Comment:
this entire logic here needs to change.
I wasn't aware how exactly caching/watchnamespace works. The issue with this is that if operator is running in target ns mode, apim won't be able to watch the cm on operator ns, and the sub controller won't be able to watch subs.
Possibly a solution is coming in the [PR](https://github.com/3scale/3scale-operator/pull/936/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R115-R119)